SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -320,12 +322,61 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
 
   bool incompressible;
   bool sustaining_terms;
+  bool axisymmetric;
 
   /*!
    * \brief A virtual member. Get strain magnitude based on perturbed reynolds stress matrix
    * \param[in] turb_ke: turbulent kinetic energy of the node
    */
   void SetPerturbedStrainMag(su2double turb_ke);
+  
+  /*!
+   * \brief Add contribution due to axisymmetric formulation to 2D residual
+   */
+  inline void ResidualAxisymmetric(su2double alfa_blended, su2double zeta){
+
+    if (Coord_i[1] < EPS) {
+      return;
+    }
+    
+    else{
+      
+      su2double yinv, rhov;
+      su2double sigma_k_i, sigma_omega_i;
+      su2double pk_axi, pw_axi, ck_axi, cw_axi, dk_axi, dw_axi;
+      
+      yinv = 1.0/Coord_i[1];
```

Comment:
```suggestion
      AD::SetPreaccIn(Coord_i[1]);
      yinv = 1.0/Coord_i[1];
```

Reply:
We would need that for the discrete adjoint, I think the coordinate is the only new input this section of code needs? Everything else is already used (and registered) in ComputeResidual.